mltsp/util.py

```diff
@@ -81,3 +82,44 @@ def is_running_in_docker():
     else:
         in_docker_container = False
     return in_docker_container
+
+
+def cast_model_params(model_type, model_params):
+    """Cast model parameter strings to expected types."""
+    from .ext.sklearn_models import model_descriptions
+    for entry in model_descriptions:
+        if entry["abbr"] == model_type:
+            params_list = entry["params"]
+            break
+    for k, v in model_params.items():
+        if v == "None":
+            model_params[k] = None
+            continue
+        for p in params_list:
+            if p["name"] == k:
+                param_entry = p
+                break
+        if type(param_entry["type"]) == type and param_entry["type"] != list:
```

Comment:
Got it. What if it was always a list of types, and sometimes it just old had one element? Then you wouldn't need separate logic for `type` and `list`.